plugins/modules/pfsense_default_gateway.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Orion Poplawski <orion@nwra.com>
+# Copyright: (c) 2018, Frederic Bor <frederic.bor@wanadoo.fr>
+# Copyright: (c) 2023, Nicolas Zagulajew <github@xoop.org>
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: pfsense_default_gateway
+version_added: 0.6.0
+author: "Nicolas Zagulajew (@freeeflyer)"
+short_description: Manage pfSense default gateway
+description: Check and update pfSense default gateway
+notes:
+options:
+  name:
+    description: Default gateway name
+    required: false
+    type: str
+  ipprotocol:
+    description: Choose the Internet Protocol Version for this gateway.
+    required: false
+    choices: [ "inet", "inet6" ]
+    default: inet
+    type: str
+  state:
+    description: present or absent will either set or unset default gateway, read will return both defaultgw4 and defaultgw6 values
+    choices: [ "present", "absent", "read" ]
+    default: read
```

Comment:
So, `read` is new, and not something I've seen used in ansible modules before.  Can you point to any similar usage?

I'm also leaning more towards simply dropping `state` altogether and just using the `None` option to express that.

Finally, we generally go with lowercase options so lets use `automatic` and `none`.

Reply:
I wanted that too.. but what to do if you just want the current default gateway ?
Also I have to rework my code. I just realized there was an issue with dynamic gateways..

Reply:
But that's just not really how ansible works.  You declare the state that you want.  If you don't want to change anything, don't use this task.

Reply:
So if I get it right.. 
- No more "state" 
- just "name" with "none" for absent 
- no read 

I could implement the fact that it accepts no option at all and return both v4 and v6 default gateway..

Question : I implemented a _get_gateway_list method because I needed it.. I feel it would be better placed in the pfsense object .. what do you think ?

Reply:
I've done it.. What do you think ?

Reply:
And lowercase is done too